EngineLayer/PeptideSpectralMatch.cs

```diff
@@ -268,6 +278,117 @@ public static int GetLongestIonSeriesBidirectional(Dictionary<PeptideWithSetModi
+                {
+                    fragmentCoveredAminoAcids.Add(this.BaseSequence.Length);
+                }
+
+                // if the first NFragment is present, first AA is covered
+                if (nTermFragmentAAPositions.Contains(1))
+                {
+                    fragmentCoveredAminoAcids.Add(1);
+                }
+
+                //Check all amino acids except for the last one in the list
+                for (int i = 0; i < nTermFragmentAAPositions.Count - 1; i++)
+                {
+                    //sequential AA, second one is covered
+                    if (nTermFragmentAAPositions[i + 1] - nTermFragmentAAPositions[i] == 1)
+                    {
+                        fragmentCoveredAminoAcids.Add(nTermFragmentAAPositions[i + 1]);
+                    }
+
+                    //check to see if the position is covered from both directions, inclusive
+                    if (cTermFragmentAAPositions.Contains(nTermFragmentAAPositions[i + 1]))
+                    {
+                        fragmentCoveredAminoAcids.Add(nTermFragmentAAPositions[i + 1]);
+                    }
+
+                    //check to see if the position is covered from both directions, exclusive
+                    if (cTermFragmentAAPositions.Contains(nTermFragmentAAPositions[i + 1] + 2))
+                    {
+                        fragmentCoveredAminoAcids.Add(nTermFragmentAAPositions[i + 1] + 1);
+                    }
+                }
+
+            }
+
+            //Check C term frags
+            if (cTermFragmentAAPositions.Any())
+            {
+                cTermFragmentAAPositions.Sort();
+
+                //if the second AA is present, the first AA is covered
+                if (cTermFragmentAAPositions.Contains(2))
+                {
+                    fragmentCoveredAminoAcids.Add(1);
+                }
+
+                //if the last AA is present, the final AA is covered
+                if (cTermFragmentAAPositions.Contains(this.BaseSequence.Length))
+                {
+                    fragmentCoveredAminoAcids.Add(this.BaseSequence.Length);
+                }
+
+                //check all amino acids except for the last one in the list
+                for (int i = 0; i < cTermFragmentAAPositions.Count - 1; i++)
+                {
+                    //sequential AA, the first one is covered
+                    if (cTermFragmentAAPositions[i + 1] - cTermFragmentAAPositions[i] == 1)
+                    {
+                        fragmentCoveredAminoAcids.Add(cTermFragmentAAPositions[i]);
+                    }
+                }
+            }
+
+            //store in PSM
+            var fragmentCoveredAminoAcidsList = fragmentCoveredAminoAcids.ToList();
+            fragmentCoveredAminoAcidsList.Sort();
+            this.FragmentCoveragePositionInPSM = fragmentCoveredAminoAcidsList;
+        }
+
+        public void GetAminoAcidCoverageProtein()
```

Comment:
add a comment that this only concerns the fragment coverage of one peptide in the protein and not the coverage of the entire protein. the method name is a little misleading.

Reply:
This method is gone now